lib/grape/validations/types/build_coercer.rb

```diff
@@ -83,7 +83,11 @@ def self.cache_instance(type, method, strict, &_block)
       end
 
       def self.cache_key(type, method, strict)
-        [type, method, strict].compact.map(&:to_s).join('_')
+        [type, method, strict].each_with_object(+'') do |val, memo|
```

Comment:
Should be `_` to avoid `fo_obar` and `foo_bar` type situations.

Reply:
I guess joining '_' is unnecessary since the key is meant for "machines" here. Do you think it is necessary? :slightly_smiling_face:

Reply:
Are we 100% sure that type/method/strict don't have the situation like `type=fo` + `method=obar` vs. `type=foo` + `method=bar`, it would generate the same key. So I think better safe than sorry.

Reply:
Good point! I've changed it.